### PR TITLE
Add oidc trust docs

### PR DESCRIPTION
**Pull request by glena**
## Description
Add OIDC trust docs, meaningful changes are under `themes/default/content/docs/pulumi-cloud/oidc`, the other changes are updates to the URLs to point to the new path

The moved pages have aliases to prevent SEO from breaking

## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: themes/default/content/docs/pulumi-cloud/oidc/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/_index.md:17]:
```suggestion
There are two ways Pulumi can integrate using OpenID Connect.  Pulumi can operate as an [OIDC provider](/docs/pulumi-cloud/oidc/provider/) issuing signed, short-lived tokens that can be exchanged by short-term credentials from your cloud provider; or as an [OIDC client](/docs/pulumi-cloud/oidc/client/) accepting short-liver OIDC Tokens issued by a trusted OIDC provided to be exchanged by short-liver Pulumi access tokens.
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:15]:
```suggestion
Pulumi supports establishing trust relationships with third party OIDC providers by leveraging id_tokens and allowing it to be exchanged for a short-lived Pulumi access token. This mechanism enhances security by eliminating the necessity for hardcoded credentials.
```

Comment (cleverguy25) [reply themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:15]: I have seen this concept of avoiding the need for "secret zero" in docs across industry. Should we allude to that for SEO purposes?

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:92]:
```suggestion
You can target the pod name by defining the path as `"kubernetes.io".pod.name`.
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (glena) [reply themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:15]: I will add something about that

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (glena) [reply themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:92]: lol I have PTSD after deployment runners, I see pools everywhere

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (glena) [reply themes/default/content/docs/pulumi-cloud/oidc/client/_index.md]: got into the buzz word train and added zero trust to the mix too, let me know what do you think

File: themes/default/content/docs/pulumi-cloud/oidc/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/_index.md:21]:
```suggestion
The integration of different pieces of infrastructure and tools comes with the unavoidable requirement of distributing credentials across all the different pieces that interact. Different mechanisms evolved to make this process simpler, but there is always a last secret to unblock secure access to a secrets vault, this is known as "Secret Zero".
```

File: themes/default/content/docs/pulumi-cloud/oidc/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/_index.md:23]:
```suggestion
Relying on OpenId Connect solves the Secret Zero problem while complying with a Zero Trust model. It establishes a secure mechanism to exchange credentials between Pulumi and other services, removing the need to maintain that last secret and reducing the risk of leaking a credential that could make the entire system vulnerable.
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:19]:
```suggestion
For third party services that have capabilities to issue OIDC id_tokens, it is possible to register them as a trusted OIDC Issuer to leverage these tokens to be exchanged by a short-lived Pulumi access token automatically to avoid having to store hardcoded credentials.
```

File: themes/default/content/docs/pulumi-cloud/oidc/_index.md
Comment (glena) [inline themes/default/content/docs/pulumi-cloud/oidc/_index.md:23]:
```suggestion
By relying on OpenId Connect we get to solve the Secret Zero problem while complying to the bases of a Zero Trust model. It stablishes a secure mechanism to exchange credentials between Pulumi and other services removing the need of securing that last secret and reducing the risk of leaking a credential that could turn the system vulnerable.
```

File: themes/default/content/docs/pulumi-cloud/oidc/_index.md
Comment (glena) [inline themes/default/content/docs/pulumi-cloud/oidc/_index.md]:
```suggestion
Relying on OpenId Connect solves the Secret Zero problem while complying with a Zero Trust model. It establishes a secure mechanism to exchange credentials between Pulumi and other services, removing the need to maintain that last secret and reducing the risk of leaking a credential that could make the entire system vulnerable.
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:64]:
```suggestion
When configuring a policy, it is required to explicitly state what kind of token can be requested and what team or user the token should be scoped to.
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/_index.md
Comment (cleverguy25) [inline themes/default/content/docs/pulumi-cloud/oidc/client/_index.md:94]:
```suggestion
Note the use of quotes to escape dots in the object keys.
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (EronWright) [inline themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:20]: Consider using this link to Kubernetes user documentation, rather than pointing to the underlying design doc.
https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (EronWright) [inline themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:17]: the other prerequisite is that you have a GKE cluster.

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (EronWright) [inline themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:26]:
```suggestion
1. Name the issuer and complete the url: `https://container.googleapis.com/v1/projects/PROJECT_NAME/locations/LOCATION/clusters/CLUSTER_NAME/`
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (EronWright) [inline themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:30]:
```suggestion
## Configure the Authorization Policies
```

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (EronWright) [inline themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:35]: Let's be sure to always include a namespace check in the rule(s).

But let's focus on granting based on the Kubernetes service account, not on the pod name.  Because that's closer to the intention of the feature; the "service account" is the "user" here.

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (EronWright) [inline themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:55]: I assumed that we would provide a CLI command for this, or enhance the login command, to do the low-level token exchange on their behalf. Is there a design doc section where we could discuss further?

Seems almost premature to talk about this feature until we have this aspect sorted out.

**Review by EronWright — CHANGES_REQUESTED** [review]

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (glena) [reply themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:57]: Eventually, we have to scope to the time we have available to ship the feature. There are tons of satellite work needed to support every use-case possible, I believe for the time being, it is a good tradeoff to provide guidelines on how this can be done and in the future replace with proper tooling support.

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (glena) [reply themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md]: done

File: themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md
Comment (EronWright) [inline themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md:29]:
```suggestion
1. Name the issuer and complete the url: `https://container.googleapis.com/v1/projects/PROJECT_NAME/locations/LOCATION/clusters/CLUSTER_NAME`
```

The trailing slash is harmless but shouldn't be there.

**Review by EronWright — APPROVED** [review]
I think it would be more secure and familiar to use a policy based on `sub` - the pod's service account identity - rather than the pod's name.

File: themes/default/content/blog/oidc-trust-relationships/index.md
Comment (arunkumar611) [inline themes/default/content/blog/oidc-trust-relationships/index.md:38]: add 'new-feature' tag. That's the tag we use for all new features.

File: themes/default/content/blog/oidc-trust-relationships/index.md
Comment (arunkumar611) [inline themes/default/content/blog/oidc-trust-relationships/index.md:54]: Change CICD -> CI/CD

**Comment by cnunciato** [comment]
Hey folks, in the future, let's please get a review from someone on the docs team before merging significant changes like these. (The path changes here have resulted in 404s that could've been prevented with an alias.) Feel free to ping us in the #docs channel if you need. Thanks!

**Pull request merged**
